g2p/api_v2.py

```diff
@@ -0,0 +1,406 @@
+                                in_lang=tr.in_lang,
+                                out_lang=tr.out_lang,
+                                substring_alignments=tier.substring_alignments(),
+                            )
+                            if request.indices:
+                                conv.input_nodes = list(tier.input_string)
+                                conv.output_nodes = list(tier.output_string)
+                                conv.alignments = tier.alignments()
+                            conversions.insert(0, conv)
+                if composed_tiers:
+                    composed_tg = CompositeTransductionGraph(composed_tiers)
+                    conv = Conversion(
+                        in_lang=request.compose_from,
+                        out_lang=transducer.out_lang,
+                        substring_alignments=composed_tg.substring_alignments(),
+                    )
+                    if request.indices:
+                        conv.input_nodes = list(composed_tg.input_string)
+                        conv.output_nodes = list(composed_tg.output_string)
+                        conv.alignments = composed_tg.alignments()
+                    conversions.insert(0, conv)
+            else:
+                for tr, tier in zip(transducer.transducers, tg.tiers):
+                    conv = Conversion(
+                        in_lang=tr.in_lang,
+                        out_lang=tr.out_lang,
+                        substring_alignments=tier.substring_alignments(),
+                    )
+                    if request.indices:
+                        conv.input_nodes = list(tier.input_string)
+                        conv.output_nodes = list(tier.output_string)
+                        conv.alignments = tier.alignments()
+                    conversions.insert(0, conv)
+        segments.append(Segment(conversions=conversions))
+    return segments
+
+
+@api.get(
+    "/outputs_for/{lang}",
+    response_description="List of language codes into which {lang} can be converted",
+)
+def outputs_for(
+    lang: LanguageNode = Path(description="Input language name"),
+) -> List[str]:
+    """Get the possible output languages for a given input language. These
+    are all the phonetic or orthographic systems into which you can convert
+    this input.
+    """
+    return sorted(descendants(g2p_langs.LANGS_NETWORK, lang.name))
+
+
+@api.get(
+    "/inputs_for/{lang}",
+    response_description="List of language codes which can be converted into {lang}",
+)
+def inputs_for(
+    lang: LanguageNode = Path(description="Output language name"),
+) -> List[str]:
+    """Get the possible input languages for a given output language. These
+    are all the phonetic or orthographic systems that you can convert
+    into this output.
+    """
+    return sorted(ancestors(g2p_langs.LANGS_NETWORK, lang.name))
+
+
+@api.get(
+    "/path/{in_lang}/{out_lang}",
+    response_description="Path from {in_lang} to {out_lang}",
+)
```

Comment:
You get slightly better documentation if you add
`response_model=List[str]` to the decorator. I think all the endpoints would get better generated documentation if you specify the `response_model` explicitly. I don't know why the function's `-> List[str]` return type declaraction doesn't already do the same thing!